lab_3/main.py

```diff
@@ -222,7 +372,24 @@ def save(self, name: str) -> int:
         :param name: name of the json file with .json format
         :return: 0 if profile saves, 1 if any errors occurred
         """
-        pass
+        if not isinstance(name, str):
+            return 1
+        decoded_freq = {}
+        profile_as_dict = {}
+        # decode_ng = ''
+        for trie in self.tries:
+            for n_gram in trie.n_gram_frequencies:
+                decode_ng = ''
+                for id_letter in n_gram:
+                    decode_ng += self.storage.get_letter_by_id(id_letter)
+                decoded_freq[decode_ng] = trie.n_gram_frequencies[n_gram]
+        profile_as_dict['name'] = self.language
+        profile_as_dict['freq'] = decoded_freq
+        profile_as_dict['n_words'] = self.n_words
+        with open(name, 'w', encoding='utf-8') as file:
+            json_string = json.dumps(profile_as_dict)
+            file.write(json_string)
```

Comment:
There is the method `.dump` in the `json` package which combines these two lines.